libgalois/include/katana/PODResizeableArray.h

```diff
@@ -139,11 +141,19 @@ class PODResizeableArray {
         capacity_ = 0;
       }
     } else if (size_ < capacity_) {
-      _Tp* new_data_ = static_cast<_Tp*>(
-          realloc(reinterpret_cast<void*>(data_), size_ * sizeof(_Tp)));
-      KATANA_LOG_DEBUG_ASSERT(new_data_);
-      data_ = new_data_;
-      capacity_ = size_;
+      if (size_ >= kMinNonZeroCapacity) {
```

Comment:
I feel like it would be simpler to write something like:
```cpp
else if (size_ < capacity_) {
  new_cap = std::max(size_, kMinNonZeroCapacity);
  new_data_ = ....

  capacity_ = new_cap;
}

```